src/dmd/traits.d

```diff
@@ -1645,7 +1645,38 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         if (cmp(s1,s2) || cmp(s2,s1))
             return True();
 
-        return (s1 == s2) ? True() : False();
+        if (s1 == s2)
+            return True();
+
+        // https://issues.dlang.org/show_bug.cgi?id=18771
+        // OverloadSets are equal if they contain the same functions
+        auto overSet1 = s1.isOverloadSet();
+        auto overSet2 = s2.isOverloadSet();
+        if (overSet1 && overSet2)
+        {
```

Comment:
Can you invert the clauses so that the return are all early returns? (this reduces indentation
e.g.
```
if (!overSet1 || !overSet2)
    return False();
else if (overSet1.a.dim != overSet2.a.dim)
    return False();
```